### ISSUE TITLE
Optional rancher2_notifier config elements set outside Terraform config not removed on terraform apply

### ISSUE THREAD

**Issue by axeal** [rancher/terraform-provider-rancher2#382]
## Type ##

Bug

## Steps to reproduce ##

1. Provision a Rancher v2.4.5 instance with single all role node custom cluster (I used https://github.com/superseb/tf-do-rancher2/)
2. Create a simple Terraform config `main.tf` per the below example:

```
provider "rancher2" {
  api_url    = "https://164.90.187.250"
  access_key = "token-l6tz2"
  secret_key = "fvvkz89c9dp6grrpn8xrjtgjw78gl42jgtkbxjch8854b5q6fttvtt"
  insecure   = true
}

resource "rancher2_notifier" "foo" {
  name          = "foo"
  cluster_id    = "c-94x2h"
  description   = "Terraform notifier acceptance test"
  send_resolved = "true"
  smtp_config {
    default_recipient = "alex@example.org"
    host              = "smtp.example.org"
    port              = 25
    sender            = "rancher@example.org"
    tls               = false
  }
}

resource "rancher2_notifier" "bar" {
  name          = "bar"
  cluster_id    = "c-94x2h"
  description   = "Terraform notifier acceptance test"
  send_resolved = "true"
  webhook_config {
    url = "example.org"
  }
}
```
3. Run `terraform init` and `terraform apply`
4. Via the UI add a username to the `foo` smtp configuration and a proxy to the bar webhook config
5. Run `terraform apply` and observe output that terraform will remove the username and proxy configs:

```
Terraform will perform the following actions:

  # rancher2_notifier.bar will be updated in-place
  ~ resource "rancher2_notifier" "bar" {
        annotations   = {}
        cluster_id    = "c-94x2h"
        description   = "Terraform notifier acceptance test"
        id            = "c-94x2h:n-kfdvp"
        labels        = {
            "cattle.io/creator" = "norman"
        }
        name          = "bar"
        send_resolved = true

      ~ webhook_config {
          - proxy_url = "http://proxy" -> null
            url       = "example.org"
        }
    }

  # rancher2_notifier.foo will be updated in-place
  ~ resource "rancher2_notifier" "foo" {
        annotations   = {}
        cluster_id    = "c-94x2h"
        description   = "Terraform notifier acceptance test"
        id            = "c-94x2h:n-qqc78"
        labels        = {
            "cattle.io/creator" = "norman"
        }
        name          = "foo"
        send_resolved = true

      ~ smtp_config {
            default_recipient = "alex@example.org"
            host              = "smtp.example.org"
            port              = 25
            sender            = "rancher@example.org"
            tls               = false
          - username          = "username" -> null
        }
    }

Plan: 0 to add, 2 to change, 0 to destroy.
```
6. Observe that the username and proxy configurations are not in fact removed, and a further `terraform apply` will still show a plan to remove them, which it does not.

**Comment by rawmind0** [comment]
Submitted PR https://github.com/rancher/terraform-provider-rancher2/pull/706 to fix this issue

**Comment by rawmind0** [comment]
PR #706 is already merged. Fix will be available at next tfp release, v1.16.1
